benchmarks/instruction_counts/definitions/standard.py

```diff
@@ -51,17 +51,79 @@
     },
 
     "Pointwise": {
-        "Math": {
-            "add": {
-                "Tensor-Scalar": GroupedStmts(
-                    r"x += 1.0",
-                    r"x += 1.0;",
-                    setup=Setup.GENERIC.value,
-                ),
-            },
-        },
+        "Math": GroupedVariants(*parse_stmts(r"""
+            Python                                   | C++
+            ---------------------------------------- | ----------------------------------------
+            # @setup                                 | // @setup
+            torch.manual_seed(138_10_23)             | torch::manual_seed(1381023);
+            x = torch.rand((4, 4))                   | auto x = torch::rand({4, 4});
+            y_float = torch.ones((4, 4))             | auto y_float = torch::ones({4, 4});
+            y_vector = torch.ones((4, 1))            | auto y_vector = torch::ones({4, 1});
+            y_int = torch.ones(                      | auto y_int = torch::ones({4, 4}, at::kInt);
+                (4, 4), dtype=torch.int32)           |
+                                                     |
+            # @add                                   | // @add
+            x += 1.0                                 | x += 1;
+            x += y_float                             | x += y_float;
+            x += y_vector                            | x += y_vector;
+            x += y_int                               | x += y_int;
+            x + y_float                              | x + y_float;
```

Comment:
Is it worth adding separate `out` variant benchmarks, e.g. `torch.add(y, y, out=x)`, `torch::add_out(x, y, y);`? I had a recent PR that only improved `out` variants and not functional/inplace, so I ended up adding an `add_out` benchmark locally to this suite to measure: https://github.com/pytorch/pytorch/pull/53575.

It might be worth measuring `out` variants more explicitly if we know that they're used frequently (e.g. in static runtime).

I also don't want to over-index on that PR though, since most of the code-paths in the functional/out variants are the same. It probably depends on how comprehensive we want the whole suit to be.

Reply:
It's a good point. I had kind of treated `x += y` as the same as `torch.add(x, y, out=x)`, but they're really not the same. (Or at least they don't have to be.) I've added both in-place and out of place versions of `torch.` / `torch::` for add and mul.